### PR TITLE
Remove key from index. Minor bug fix.

### PR DESCRIPTION
**Pull request by adlrocha**
This PR: 
- Implements phase 1 of the `remove` implementation described in #5 to enable the implementation of storage interface in storetheindex https://github.com/filecoin-project/storetheindex/issues/7
- Fixes and error that caused the freelist to not to `Flush()` correctly with the rest of the storage.

Coming next: 
- Compaction
- https://github.com/ipld/go-storethehash/issues/7

//cc @gammazero

### CODE REVIEW

File: store/store.go
Comment (gammazero) [inline store/store.go:310]: While it does not need to block this PR, I would like to understand the design behind the flush mechanism and rate limiting.  It looks like `flushTick` does a sleep if the previous operation was sufficiently recent.  Why is that needed, instead of letting backpressure from the underlying storage provide rate limiting?

File: store/store_test.go
Comment (gammazero) [inline store/store_test.go:35]: Note that the `testing` package provides its own `TempDir` function: https://pkg.go.dev/testing@go1.16.5#T.TempDir

**Review by gammazero — APPROVED** [review]
I have a few questions about how this is supposed to work.  I am not sure if my review is complete without that understanding, but I do not know if it is work holding up this PR.

So, take a look at my comments/questions and decide if we want to review the flush/sync mechanism before this PR gets merged, or if that is better done later.

File: store/store.go
Comment (adlrocha) [reply store/store.go:285]: The key stored in primary storage is in its original form (in our case a CID). According to the primary storage used, `IndexKeys` may be transformed into another key space (for the CIDPrimary, a multihash). What this line does is converting `storedKey` into the same space as the input key to compare them (and avoid two CIDs for the same multihash from being interpreted as different keys).

File: store/index/index.go
Comment (adlrocha) [reply store/index/index.go:381]: `GetRecord` can only return `nil` or the actual record. If there is a record for that key in the recordlist it returns it, if not, it returns `nil`. What `PutKeys` with an empty key is doing is removing all records from `r.Pos` to `r.NextPos`, i.e. the position in the recordlist of the record we are removing.

I think it is OK to just check if it is `nil`.

File: store/store.go
Comment (adlrocha) [reply store/store.go:310]: This is legacy code from @hannahhoward's initial design. Now that you mention it, it would be great to get @hannahhoward's input on this. I will open a new issue in order not to block this PR. Thanks.

**Pull request merged**
